INSTALL.md

```diff
@@ -25,6 +25,50 @@ dependencies, as used for the binary releases, by using the
 [depends README](depends/README.md) for instructions to build Dogecoin using
 these dependencies.
 
+### Docker installation
+
+Host your node in a container ! To use it, first follow their [Get Docker](https://docs.docker.com/get-docker/) instructions to install Docker on your computer.
+
+For security reasons, to avoid running `docker` commands with `sudo` rights, add your user in the `docker` group. See [Manage Docker as a non-root user](https://docs.docker.com/engine/install/linux-postinstall/).
```

Comment:
Spelling mistake: 
> `docker` command

Reply:
so last night i checked out your docker branch and encountered the following error:

```
man: /dogecoin/share/man/man1/dogecoind.1: No such file or directory
No manual entry for /dogecoin/share/man/man1/dogecoind.1
Traceback (most recent call last):
  File "/entrypoint.py", line 126, in <module>
    EXECUTABLE_ARGS = convert_env()
  File "/entrypoint.py", line 91, in convert_env
    dogecoind_options = executable_options()
  File "/entrypoint.py", line 42, in executable_options
    raw_options = subprocess.check_output(
  File "/usr/lib/python3.9/subprocess.py", line 424, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'bash -c "man /dogecoin/share/man/man1/dogecoind.1 | grep '^ *-'"' returned non-zero exit status 1.
```

just a couple notes mostly for Dockerfile which i see you've already addressed for the most part. i'd set version to 1.14.4 for now since there are no binaries available for 1.14.5-dev. unless ofc you are waiting for 1.14.5 release to include these changes.

i don't think you need the adduser lines as dogecoind can be managed by root within the container, don't need python3 in ENTRYPOINT line and depending on what you choose to do with your entrypoint, a more simple path would be adding CMD['dogecoind'] at the bottom. also need some more work, addl libs to connect with host display adapter, to enable dogecoin-qt to work.

outside of that i appreciate everything you're doing. ♥️

Reply:
> i don't think you need the adduser

It's a security practice to create a user, it can be recommended, by [OWASP](https://cheatsheetseries.owasp.org/cheatsheets/Docker_Security_Cheat_Sheet.html) for example. If an intrusion occur through your exposed application, the attacker will have only application/user rights, and not root rights. If it has root rights within the container, the interaction with other container will be easier. Container is a jail, you should create a user.

About `dogecoind` as an entry point, there is an issue with right permission of `datadir` and volumes. When you create your container, you need to create `datadir` folder from inside the running container, which is theoretically with the right of your created users for the previous security reason. But with those right, you can't create directly a folder inside the host folder, you need root rights of your container, which are available at the build time of the image. With the `USER` instruction inside the build, you won't have root permission inside a running container, so you need to do a trick with `gosu`.
A lot of small details about docker permissions :)

For QT, yeah would be great improvement to have this, if anyone has some skills about X server, help is welcome :)

For your error, I will investigate :mag:  Did you still have the command who created this ?

Reply:
i followed instructions in INSTALL.md

Reply:
Weird... Looks like it works from my side. It seems strange, it says you don't have man which I'm able to get with a default setup. Did you use a volume or not at all, strictly the instruction of the `INSTALL.md` ?

Because it may not be your error, but if you're doing a volume with the home like `-v $(pwd)/data:/dogecoin`, in this case you will remove the folder `/dogecoin/share/man/...` who contain man pages, and this will break the script.

Need to move those man pages.

Reply:
interesting. it's my fault for the error.  i had omitted --strip-components=1 in that i was under the impression as indicated below that the binaries extracted from tar.gz were stripped:
```
~/dogecoin-1.14.4/bin$ file dogecoind
dogecoind: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.24, BuildID[sha1]=ac086eecfffb8c7aaa90945015c2e0789300226c, stripped
file dogecoin-cli
~/dogecoin-1.14.4/bin$ dogecoin-cli: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.24, BuildID[sha1]=cb480a196bfae9a2c764ecee74baf2ce61c81fa0, stripped
~/dogecoin-1.14.4/bin$ file dogecoin-qt
dogecoin-qt: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.24, BuildID[sha1]=78fa3a3cf4c3c80fbaf48e1ad8b58fe8a25d8cfd, stripped
~/dogecoin-1.14.4/bin$ file test_dogecoin 
test_dogecoin: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.24, BuildID[sha1]=050fd6c34407fee30141f2bd245bc1a8da76abb9, stripped
~/dogecoin-1.14.4/bin$ file dogecoin-tx
dogecoin-tx: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.24, BuildID[sha1]=9c7a0e58abecb3db122931bf6951bcc7e632d158, stripped
```

can confirm that it works as is without the omission of --strip-components=1. my mistake :)

Reply:
Just saw your work https://github.com/xanimo/dogecoin-core, nice to see you're Dockerfile hobbyist too :)

## Binaries and man pages directory fix
*See https://github.com/dogecoin/dogecoin/pull/2337/commits/acd6539f07dfd8d778bef798f7127d237cb295a9*.

So your mistake bring me to interesting problems, I had to move both binaries and man pages to more adapted linux directories, `/usr/local/bin` and `/usr/share/man/man1` to avoid issue with a misconfigured volume.

I also setuid `dogecoin*` executables to always run it as dogecoin user. `docker exec`, "because" of `gosu` and no `USER` field specified in the Dockerfile, is using root as default user to run commands. It's to switch from root to dogecoin rights at each `dogecoind*` commands runned with `docker exec` on a running container, instead of running `docker exec --user dogecoin [command]`.

It raises some limits, like a [process can switch from EUID to RUID](https://unix.stackexchange.com/questions/479135/how-can-i-prevent-a-process-to-change-its-setuid-bit-back-to-ruid-within-the-pro), but I suppose it increase security by restricting rights of running Dogecoin executable.

I can explain more if someone wants details, and if a Linux/unix veteran can give a feedback it would be great !

Reply:
yes indeed and you too :3, and yes you got me was sort of for personal exploration purposes, but yeah the gosu (debian) and corresponding su-exec (alpine) entrypoint files were from that repo in addition to README which i need to rewrite specific to my repo (for x-compile alpine musl) but regardless at this point it's more so an amalgam of multiple other sources, zombified if you will. spent hours going out of my mind haha. 

>It raises some limits, like a process can switch from EUID to RUID, but I suppose it increase security by restricting rights of running Dogecoin executable.

i am no linux/unix wizard but i think that's the beauty of those aforementioned files paired with this line in Dockerfile: `RUN ["chmod", "+x", "/entrypoint.sh"]`

anyway everything you've done looks great to me! looking forward to seeing finished product :)